src/o365/spo/commands/list/list-webhook-add.ts

```diff
@@ -0,0 +1,239 @@
+import auth from '../../SpoAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import * as request from 'request-promise-native';
+import {
+  CommandOption,
+  CommandValidate
+} from '../../../../Command';
+import SpoCommand from '../../SpoCommand';
+import Utils from '../../../../Utils';
+import { Auth } from '../../../../Auth';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+const expirationDateTimeMaxDays = 180;
+const maxExpirationDateTime: Date = new Date();
+// 180 days from now is the maximum expiration date for a webhook
+maxExpirationDateTime.setDate(maxExpirationDateTime.getDate() + expirationDateTimeMaxDays);
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  listId?: string;
+  listTitle?: string;
+  notificationUrl: string;
+  expirationDateTime?: string;
+  clientState?: string;
+}
+
+class SpoListWebhookSetCommand extends SpoCommand {
```

Comment:
Following the naming convention, the class name should be `SpoListWebhookAddCommand`